components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/AbstractRoleManagementListener.java

```diff
@@ -246,11 +246,22 @@ public void preGetRolesCount(String tenantDomain) throws IdentityRoleManagementE
 
     }
 
+    @Override
+    public void preGetRolesCount(String searchFilter, String tenantDomain) throws IdentityRoleManagementException {
+
+    }
+
```

Comment:
The AbstractRoleManagementListener.java abstract class is designed to serve as a base class for custom implementations of RoleManagementListener. It allows developers to override and customize only the specific methods they need, without requiring them to implement all the methods of the RoleManagementListener interface.